css/src/components/form/input.scss

```diff
@@ -0,0 +1,102 @@
+$input-color: $text !default;
+$input-background-color: $body-background !default;
+$input-border-color: $text !default;
+$input-placeholder-color: $text-subtle !default;
+
+$input-hover-color: $text !default;
+$input-hover-border-color: $primary !default;
+$input-focus-border-color: $primary !default;
+
+$input-disabled-color: $text-subtle !default;
+$input-disabled-background-color: $body-background-medium !default;
+$input-disabled-border-color: $table-border-dark !default;
+
+$input-error-border-color: $danger !default;
+$input-warning-border-color: $warning-active !default;
+$input-success-border-color: $score-high !default;
+
+$input-focus-box-shadow-size: 0 0 0 0.125rem !default;
+
+.input {
+	@include control;
+
+	width: 100%;
+	max-width: 100%;
+	border-color: $input-border-color;
+	background-color: $input-background-color;
+	color: $input-color !important;
+	margin-block-start: 0.5rem;
+
+	&::placeholder {
+		color: $input-placeholder-color;
+	}
+
+	&:hover {
+		border-color: $input-hover-border-color;
+	}
+
+	&[disabled] {
+		border-color: $input-disabled-border-color;
+		background-color: $input-disabled-background-color;
+		color: $input-disabled-color;
+	}
+
+	@include focus-visible {
+		border-color: $input-focus-border-color;
+
+		@if ($focus-visible-use-polyfill) {
+			outline: none !important;
+			box-shadow: $input-focus-box-shadow-size $input-focus-border-color;
+		} @else {
+			@extend %focus;
+
+			outline-color: $input-focus-border-color;
+			outline-offset: 0 !important;
+			outline-style: solid !important;
+		}
+	}
```

Comment:
@wibjorn I need your advice. I have two ideas on how to proceed with focus styles since we want to keep Safari looking nice. This is a first approach that keeps `focus-visible` mixin unchanged, however, I need to use the `if ... else` statement here to pass the correct CSS rules I want to be applied to the component. 

Th second approach I was thinking of is modifying the mixin like that:
```css
@mixin focus-visible($focus-color: $primary, $shadow-size: 0 0 0 0.125rem) {
	@if ($focus-visible-use-polyfill) {
		&.focus-visible {
			@content;
			outline: none !important;
			box-shadow: $shadow-size $focus-color;
		}
	} @else {
		&:focus-visible {
			@extend %focus;

			@content;
		}
	}
}
```

and then the in `input.scss` file code would look like that:

```css
@include focus-visible($input-focus-border-color, $input-focus-box-shadow-size) {
	border-color: $input-focus-border-color;
	outline-color: $input-focus-border-color;
	outline-offset: 0 !important;
	outline-style: solid !important;
}
```

Personally, I like the 2nd better, but I'm not sure if the box-shadow styles would look good on future components that might need using that mixin, including the `toggle` you're working on. Please let me know what you think.

Reply:
To emulate `focus-visible` styles like in Safari follow these steps:
1. Update `$focus-visible-use-polyfill: false` to be `true` in `tokens/focus.scss` file
2. `npm start`
3. Manually add `focus-visible` class to the `input`/`textarea` components to see how it look like

Reply:
chatted offline online, figured this out